module/entities/helper.js

```diff
@@ -1338,6 +1338,117 @@ module.exports = class EntitiesHelper {
             }
         })
     }
+
+
+    /**
+ * Update user roles in entities elastic search 
+ * @method
+ * @name updateUserRolesInEntitiesElasticSearch
+ * @name userRoles - array of userRoles.
+ * @name userId - user id
+ * @returns {Object} 
+ */
+    static updateUserRolesInEntitiesElasticSearch(userRoles = [], userId = "") {
+        return new Promise(async (resolve, reject) => {
+            try {
+            
+            await Promise.all(userRoles.map( async role => {
+                await Promise.all(role.entities.map(async entity => {
+
+                    let entityDocument = await elasticSearch.get
+                    (
+                        entity,
+                        process.env.ELASTICSEARCH_ENTITIES_INDEX
+                    )
+                   
+                    if (entityDocument.statusCode == httpStatusCode.ok.status) {
+
+                        entityDocument = entityDocument.body["_source"].data;
+
+                        if (entityDocument.roles[role.code]) {
+                            if (!entityDocument.roles[role.code].includes(userId)) {
+                                entityDocument.roles[role.code].push(userId);
+                            }
```

Comment:
@deepakr123 - We should not update if its not required.

Reply:
resolved